### PR TITLE
perf: fix cache localName to speed up transpilation

### PR DESCRIPTION
**Pull request by robin-drexler**
Please see: https://github.com/styled-components/babel-plugin-styled-components/issues/158#issuecomment-418257001

### CODE REVIEW

File: src/utils/detectors.js [hunk 1]
```diff
@@ -8,7 +8,16 @@ const VALID_TOP_LEVEL_IMPORT_PATHS = [
 export const isValidTopLevelImport = x =>
   VALID_TOP_LEVEL_IMPORT_PATHS.includes(x)
 
+
+const localNameCache = {};
+
 const importLocalName = (name, state) => {
+  const cacheKey = name + state.file.opts.filename;
```
Comment (mxstbr) [inline src/utils/detectors.js]: Is this unique enough? Is there anything else we need to take into account?

**Comment by robin-drexler** [comment]
I didn't. TBH, I don't have a repository yet that includes loads of styled components. We're just getting started. So I don't know how to actually proof it still works.

**Comment by mxstbr** [comment]
If you publish a dev version I can try it on Spectrum

**Comment by mxstbr** [comment]
Tried building Spectrum with this just now.

- Build time with `@latest`: 190s
- Build time with `@dev`: 50s!!!!!!!!!

Don't know if it breaks anything but from a perf perspective I'm 👍

**Comment by probablyup** [comment]
@mxstbr I mean if things visually looked okay to you, I'm fine with merging

**Comment by mxstbr** [comment]
One sec double checking and merging if that's the case

**Comment by mxstbr** [comment]
Looks fine to me: http://spectrum-gyuzcdwafh.now.sh/ :+1:

Let's shipit

**Comment by mxstbr** [comment]
Thank you so much for helping us improve styled-components! Based on our [Community Guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md) every person that has a PR of any kind merged is offered **an invitation to the styled-components organization**—that is you right now!

Should you accept, you'll get write access to the main repository. (and a fancy `styled-components` logo on your GitHub profile!) You'll be able to label and close issues, merge pull requests, create new branches, etc. We want you to help us out with those things, so don't be scared to do them! Make sure to read our [contribution and community guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md), which explains all of this in more detail. Of course, if you have any questions just let me know!

**Pull request merged**